pkg/apis/events/v1alpha1/cloudbuildsource_validation.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	duckv1alpha1 "github.com/google/knative-gcp/pkg/apis/duck/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+func (current *CloudBuildSource) Validate(ctx context.Context) *apis.FieldError {
+	errs := current.Spec.Validate(ctx).ViaField("spec")
+	return duckv1alpha1.ValidateAutoscalingAnnotations(ctx, current.Annotations, errs)
+}
+
+func (current *CloudBuildSourceSpec) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+
+	// Topic [optional], if specified , it has to equal to the "cloud-builds"
+	if current.Topic == "" {
+		errs = errs.Also(apis.ErrMissingField("topic"))
+	}
+
+	if current.Topic != DefaultTopic {
```

Comment:
OK, this confirms that we shouldn't add topic as a configurable thing in the spec